en/modules/modules_system.md

```diff
@@ -945,3 +945,50 @@ work_queue <command> [arguments...]
 
    status        print status info
 ```
+
+## internal_combustion_engine_control
+Source: [modules/internal](https://github.com/PX4/PX4-Autopilot/tree/main/src/modules/internal_combustion_engine_control)
+
+**NOTE:** This feature is not enabled by default and needs to be configured with the rpm capture driver.
+
+### Description
+The module controls the internal combustion engine ICE as described in the following architecture diagram:
+
+![Architecture](../../assets/diagrams/ice_control_diagram.png)
+
+The module publishes `InternalCombustionEngineControl.msg` containing:
```

Comment:
In the source you can link to this if you like. It will be at [InternalCombustionEngineControl.msg](../msg_docs/InternalCombustionEngineControl.md)